torch_frame/data/multi_embedding_tensor.py

```diff
@@ -0,0 +1,166 @@
+        return MultiEmbeddingTensor(self.num_rows, self.num_cols,
+                                    self.values.clone(), self.offset.clone())
+
+    def __repr__(self) -> str:
+        return ' '.join([
+            f"{self.__class__.__name__}(num_rows={self.num_rows},",
+            f"num_cols={self.num_cols},",
+            f"device='{self.device}')",
+        ])
+
+    # Device Transfer #########################################################
+
+    def to(self, *args, **kwargs):
+        return self._apply(lambda x: x.to(*args, **kwargs))
+
+    def cpu(self, *args, **kwargs):
+        return self._apply(lambda x: x.cpu(*args, **kwargs))
+
+    def cuda(self, *args, **kwargs):
+        return self._apply(lambda x: x.cuda(*args, **kwargs))
+
+    # Helper Functions ########################################################
+
+    def _apply(self, fn: Callable[[Tensor], Tensor]) -> 'MultiEmbeddingTensor':
+        out = copy.copy(self)
+        out.values = fn(out.values)
+        out.offset = fn(out.offset)
+        return out
+
+    # Properties ##############################################################
+
+    @property
+    def ndim(self) -> int:
+        return 3
+
+    def dim(self) -> int:
+        return self.ndim
+
+    def size(self, dim: int) -> int:
+        r"""Dimension of the :class:`torch_frame.data.MultiEmbeddingTensor`"""
+        # Handle negative dim
+        if dim < 0:
+            dim = self.ndim - dim
+
+        if dim == 0:
+            return self.num_rows
+        elif dim == 1:
+            return self.num_cols
+        elif dim == 2:
+            raise ValueError(
+                "MultiEmbeddingTensor does not have a fixed length on the"
+                "third dimension.")
+        else:
+            raise IndexError(
+                "Dimension out of range (expected to be in range of [0, 2],"
+                f" but got {dim}")
+
+    @property
+    def shape(self) -> Tuple[int, int, int]:
+        return (self.num_rows, self.num_cols, -1)
+
+    @property
+    def dtype(self) -> torch.dtype:
+        return self.values.dtype
+
+    # Static methods ##########################################################
+    @staticmethod
+    def cat(
+        xs: Sequence['MultiEmbeddingTensor'],
+        dim: int = 0,
+    ) -> 'MultiEmbeddingTensor':
+        if len(xs) == 0:
+            raise RuntimeError("Cannot concatenate a list of length 0.")
+        assert isinstance(xs[0], MultiEmbeddingTensor)
+        dim = MultiEmbeddingTensor._check_dim(dim)
+        device = xs[0].device
+        if dim == 0:
+            num_rows = sum(x.num_rows for x in xs)
+            num_cols = xs[0].num_cols
+            for x in xs[1:]:
+                if x.num_cols != num_cols:
+                    raise RuntimeError(
+                        "num_cols must be the same across a list of input "
+                        "multi nested tensors.")
+            values = torch.cat([x.values for x in xs], dim=0)
```

Comment:
You probably want to check if all items in xs are in the same dtype. If not, probably good to raise an error or a warning.

Reply:
Addressing via #184